datafusion/src/execution/context.rs

```diff
@@ -132,32 +134,34 @@ use super::{
 /// # use datafusion::error::Result;
 /// # #[tokio::main]
 /// # async fn main() -> Result<()> {
-/// let mut ctx = ExecutionContext::new();
+/// let mut ctx = SessionContext::new();
 /// ctx.register_csv("example", "tests/example.csv", CsvReadOptions::new()).await?;
 /// let results = ctx.sql("SELECT a, MIN(b) FROM example GROUP BY a LIMIT 100").await?;
 /// # Ok(())
 /// # }
 /// ```
 #[derive(Clone)]
-pub struct ExecutionContext {
+pub struct SessionContext {
+    /// Uuid for the session
+    pub session_id: String,
     /// Internal state for the context
-    pub state: Arc<Mutex<ExecutionContextState>>,
+    pub state: Arc<Mutex<SessionState>>,
```

Comment:
it looks like there is an invariant between session_id and state.session_id, if so, it's better to keep `session_id` private so it is always in sync with the id in state. we can add a `session_id` method to make the id reader to users.

Reply:
> it looks like there is an invariant between session_id and state.session_id, if so, it's better to keep `session_id` private so it is always in sync with the id in state. we can add a `session_id` method to make the id reader to users.

Done.